src/number_theory/kummer_dedekind.lean

```diff
@@ -52,20 +52,157 @@ with a formula).
 kummer, dedekind, kummer dedekind, dedekind-kummer, dedekind kummer
 -/
 
+variables (R : Type*) {S : Type*} [comm_ring R] [comm_ring S] [algebra R S]
+
+open ideal polynomial double_quot unique_factorization_monoid algebra ring_hom
+
+local notation R `[` x `]` := adjoin R ({x} : set S)
```

Comment:
I am not sure, but since we now have `R[X]` for polynomials (even if it's not opened here), this can be a little confusing. But I agree we need a good notation for `adjoin`.

Reply:
I'll try and find something else!

Reply:
Note that intermediate_field.adjoin has [this](https://leanprover-community.github.io/mathlib_docs/field_theory/adjoin.html#notation) weird Unicode [notation](https://github.com/leanprover-community/mathlib/blob/e623a7e43f627d9939114c1b87b38d452911a66a/src/field_theory/adjoin.lean#L351).

Reply:
I've copied the notation for `intermediate_field.adjoin` as suggested. I'm not sure how doable that would be, but it sure would be nice if the notation for `intermediate_field.adjoin` came directly from algebra.adjoin`.